src/components/asset-list/RecyclerAssetList2/FastComponents/FastBalanceCoinRow.tsx

```diff
@@ -0,0 +1,320 @@
+import { CoinIconIndicator } from '../../../../components/coin-icon';
+import { Icon } from '../../../../components/icons';
+import { ButtonPressAnimation } from '../../../animations';
+
+import { initialChartExpandedStateSheetHeight } from '../../../expanded-state/asset/ChartExpandedState';
+import { ExtendedState } from '../core/RawRecyclerList';
+import FastCoinIcon from './FastCoinIcon';
+import { Text } from '@rainbow-me/design-system';
+import {
+  useAccountAsset,
+  useCoinListFinishEditingOptions,
+} from '@rainbow-me/hooks';
+import Routes from '@rainbow-me/routes';
+import { borders, colors, padding, shadow } from '@rainbow-me/styles';
+
+interface CoinCheckButtonProps {
+  isHidden: boolean;
+  isPinned: boolean;
+  onPress: () => void;
+  uniqueId: string;
+  theme: any;
+}
+
+const CoinCheckButton = React.memo(function CoinCheckButton({
+  isHidden,
+  isPinned,
+  onPress,
+  uniqueId,
+  theme,
+}: CoinCheckButtonProps) {
+  const { selectedItems } = useCoinListFinishEditingOptions();
+  const selected = selectedItems.includes(uniqueId);
+
+  const showOutline = !(isHidden || isPinned);
+  const coinIconPlaceholder = !selected && (isHidden || isPinned);
+
+  const checkmarkBackgroundDynamicStyle = {
+    ...shadow.buildAsObject(
+      0,
+      4,
+      12,
+      theme.isDarkMode ? colors.shadow : colors.appleBlue,
+      0.4
+    ),
+  };
+
+  return (
+    <View style={cx.checkboxContainer}>
+      <ButtonPressAnimation onPress={onPress}>
+        <View style={cx.checkboxInnerContainer}>
+          {showOutline && (
+            <View
+              style={[
+                cx.circleOutline,
+                { borderColor: colors.alpha(colors.blueGreyDark, 0.12) },
+              ]}
+            />
+          )}
+
+          {coinIconPlaceholder && (
+            <CoinIconIndicator
+              // @ts-expect-error
+              isPinned={isPinned}
+              style={cx.coinIconIndicator}
+              theme={theme}
+            />
+          )}
+
+          {selected && (
+            <View
+              style={[cx.checkmarkBackground, checkmarkBackgroundDynamicStyle]}
+            >
+              <Icon color="white" name="checkmark" />
+            </View>
+          )}
+        </View>
+      </ButtonPressAnimation>
+    </View>
+  );
+});
+
+const formatPercentageString = (percentString?: string) =>
+  percentString ? percentString.split('-').join('- ') : '-';
+
+interface MemoizedBalanceCoinRowProps {
+  uniqueId: string;
+  nativeCurrency: string;
+  theme: any;
+  navigate: any;
+  nativeCurrencySymbol: string;
+  isHidden: boolean;
+  maybeCallback: React.RefObject<null | (() => void)>;
+}
+
+const MemoizedBalanceCoinRow = React.memo(
+  ({
+    uniqueId,
+    nativeCurrency,
+    theme,
+    navigate,
+    nativeCurrencySymbol,
+    isHidden,
+    maybeCallback,
+  }: MemoizedBalanceCoinRowProps) => {
+    const item = useAccountAsset(uniqueId, nativeCurrency);
```

Comment:
This should not be here, but we'll leave this for now, it will be less expensive once we have #3196 merged

Reply:
can you add TODO?

Reply:
^^